### PR TITLE
{pkg/trace/api,pkg/trace/agent}: container tagging

### PR DESCRIPTION
**Pull request by gbbr**
This change adds support for container tagging. When the incoming
payload contains the header "Datadog-Container-ID", its value is used to
obtain orchestrator tags from the "tagger" service (pkg/tagger). These
tags are then set on each root span as tags `orchestrator.<tag> = <value>`.

### CODE REVIEW

File: pkg/trace/api/api.go [hunk 1]
```diff
@@ -294,12 +316,29 @@ func (r *HTTPReceiver) handleTraces(v Version, w http.ResponseWriter, req *http.
...
+	EntityTags map[string]string
+
+	// Spans holds the spans of this trace.
+	Spans pb.Trace
+}
+
+func (r *HTTPReceiver) processTraces(ts *info.TagStats, entityID string, traces pb.Traces) {
 	defer timing.Since("datadog.trace_agent.internal.normalize_ms", time.Now())
+
+	entityTags := getEntityTags(entityID)
```
Comment (palazzem) [inline pkg/trace/api/api.go]: Are we making this call even if the entity id is an empty value?

Comment (gbbr) [reply pkg/trace/api/api.go]: Yes. If it's empty, `tagger.Tag` returns an error, in which case `entityTags` will be an empty map.

Comment (palazzem) [reply pkg/trace/api/api.go]: OK so `tagger.Tag` will not make a real call but will just say "it's empty I will not do anything" right?

Comment (gbbr) [reply pkg/trace/api/api.go]: https://github.com/DataDog/datadog-agent/blob/e5b2695dacf0cbe25e066cfe7fd58f30daaf7b85/pkg/tagger/tagger.go#L225-L227

File: pkg/trace/api/api.go [hunk 2]
```diff
@@ -294,12 +315,29 @@ func (r *HTTPReceiver) handleTraces(v Version, w http.ResponseWriter, req *http.
 			r.wg.Done()
 			watchdog.LogOnPanic()
 		}()
-		r.processTraces(ts, traces)
+		entityID := req.Header.Get(headerEntityID)
```
Comment (ahmed-mez) [inline pkg/trace/api/api.go]: we will need to add the `kubernetes_pod://` prefix to `entityID` as in tagger we're converting pod UIDs into entity names
https://github.com/DataDog/datadog-agent/blob/master/pkg/tagger/collectors/kubelet_extract.go#L111
https://github.com/DataDog/datadog-agent/blob/master/pkg/util/kubernetes/kubelet/kubelet_common.go#L26

File: pkg/trace/api/api.go (hunk 1 above)
Comment (gbbr) [reply pkg/trace/api/api.go]: Resolving.

File: tasks/trace_agent.py [hunk 3]
```diff
@@ -11,7 +11,12 @@
 from .go import deps
 
 BIN_PATH = os.path.join(".", "bin", "trace-agent")
-DEFAULT_BUILD_TAGS = ["netcgo", "secrets"]
+DEFAULT_BUILD_TAGS = [
+    "netcgo",
```
Comment (gbbr) [inline tasks/trace_agent.py]: We might need more here, just like https://github.com/DataDog/datadog-agent/blob/29a972e3c1fdce1246105c630f74302f9678fb53/tasks/agent.py#L25-L45. Let's consolidate on that before merging.

Comment (ahmed-mez) [reply tasks/trace_agent.py]: we're only missing the `docker` build tag I think, required by the `tagger`

Comment (gbbr) [reply tasks/trace_agent.py]: Added.

**Review by hkaj — COMMENTED** [review]
Given we're only working with container IDs for entity ID here, and that the header is explicitly container-id, it makes sense to replace entity* with container* imho (EntityTags --> ContainerTags for example, which is also more precise). Container tags include Orchestrator tags anyway.

Except for this nitpick, LGTM

**Comment by gbbr** [comment]
Thanks @hkaj. Nitpicks or not, these things are important (at least to me). Changed throughout.

**Review by hkaj — DISMISSED** [review]

**Review by ahmed-mez — DISMISSED** [review]

**Review by hkaj — DISMISSED** [review]

**Review by ahmed-mez — DISMISSED** [review]

**Review by ahmed-mez — APPROVED** [review]

**Review by hkaj — APPROVED** [review]
🚢 it

**Pull request merged**